geotrek/feedback/models.py

```diff
@@ -98,19 +98,20 @@ class Meta:
         verbose_name_plural = _("Reports")
         ordering = ["-date_insert"]
 
+    @property
+    def label(self):
+        return f"{_('Report')} n°{self.pk}"
+
     def __str__(self):
-        if self.email:
-            return self.email
-        return "Anonymous report"
+        return self.label
 
     @property
-    def email_display(self):
-        return '<a data-pk="%s" href="%s" title="%s" >%s</a>' % (
-            self.pk,
-            self.get_detail_url(),
-            self,
-            self,
-        )
+    def id_display(self):
```

Comment:
for mapentity compatibility you should not customize id display but add other named property